### ISSUE TITLE
The difference between Deit and Vit in your backbone code 

### ISSUE THREAD

**Issue by yjh576** [heshuting555/TransReID#9]
Sorry, I have not found their difference in your backbone code. In fact, they should have some differences.

**Comment by heshuting555** [comment]
Deit has one more distill token compared with vit, and the difference in pretrained model loading
https://github.com/heshuting555/TransReID/blob/5901746360fc5c02b415f4dafb191d8d20d6e1c2/model/backbones/vit_pytorch.py#L425
and model configuration
https://github.com/heshuting555/TransReID/blob/5901746360fc5c02b415f4dafb191d8d20d6e1c2/model/backbones/vit_pytorch.py#L470

**Comment by yjh576** [comment]
According to the code of Deit, I think Deit's code should be:
    def forward(self, x):
        x, x_dist = self.forward_features(x)
        x = self.head(x)
        x_dist = self.head_dist(x_dist)
        if self.training:
            return x, x_dist
        else:
            # during inference, return the average of both classifier predictions
            return (x + x_dist) / 2

the above code have some differences with your TransReID:

        cls_tokens = self.cls_token.expand(B, -1, -1)  # stole cls_tokens impl from Phil Wang, thanks
        x = torch.cat((cls_tokens, x), dim=1)

        if self.cam_num > 0 and self.view_num > 0:
            x = x + self.pos_embed + self.sie_xishu * self.sie_embed[camera_id * self.view_num + view_id]
        elif self.cam_num > 0:
            x = x + self.pos_embed + self.sie_xishu * self.sie_embed[camera_id]
        elif self.view_num > 0:
            x = x + self.pos_embed + self.sie_xishu * self.sie_embed[view_id]
        else:
            x = x + self.pos_embed

        x = self.pos_drop(x)

        if self.local_feature:
            for blk in self.blocks[:-1]:
                x = blk(x)
            return x

        else:
            for blk in self.blocks:
                x = blk(x)

            x = self.norm(x)

            return x[:, 0]

    def forward(self, x, cam_label=None, view_label=None):
        x = self.forward_features(x, cam_label, view_label)
        return x

TransReID seems to lacks  x_dist.

**Comment by yjh576** [comment]
I see. You mean that TranReID only uses deit pretrained model parameters, not using knowledge distillation as Deit. Do you try to add knowledge distillation into you TransReid? It seems to be useful.

**Comment by heshuting555** [comment]
No, because 'x_dist'  has no proper supervision signal.
